### PR TITLE
Add renderToString method

### PR DESCRIPTION
**Pull request by eddyerburgh**
- Add a `renderToString` (https://github.com/vuejs/vue-test-utils/issues/427)

### CODE REVIEW

File: src/renderToString.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import Vue from 'vue'
+import createInstance from './lib/create-instance'
+import './lib/polyfills/object-assign-polyfill'
+import { throwError } from './lib/util'
+
+Vue.config.productionTip = false
+Vue.config.devtools = false
+
+export default function renderToString (component: Component, options: Options = {}): string {
+  const renderer = require('vue-server-renderer').createRenderer()
```
Comment (ematipico) [inline src/renderToString.js]: Is there a reason why `vue-server-renderer` gets imported using CommonJS and not ESM?

Comment (eddyerburgh) [reply src/renderToString.js]: ESM imports are hoisted to the top, so any ESM import will always be bundled.

vue-server-renderer relies on node modules, like path, which makes it difficult to compile with webpack. require is only run when the code runs, so putting it inside the renderToString means we only try and import vue-server-renderer when the renderToString function is run.

We should also add a warning that checks if the user is in node, and we can throw an error if they aren't to tell them renderToString needs to be run in node.

Comment (ematipico) [reply src/renderToString.js]: Clear explanation, thank you very much!

**Comment by dotnetCarpenter** [comment]
I just did a `npm update` and got *npm WARN @vue/test-utils@1.0.0-beta.12 requires a peer of vue-server-renderer@2.x but none is installed. You must install peer dependencies yourself.*.

Could you please update the [release notes](https://github.com/vuejs/vue-test-utils/releases/tag/v1.0.0-beta.12) to explain that we need to install [vue-server-renderer](https://github.com/vuejs/vue/tree/dev/packages/vue-server-renderer#readme) because of the added `renderToString method`.

I did a `git blame` on the `package.json` to see why but I imagine most developers won't go through the trouble.

**Comment by eddyerburgh** [comment]
@dotnetCarpenter thanks for finding this.

I've created an issue to remove it as a peerDependency—https://github.com/vuejs/vue-test-utils/issues/440.

I think most users won't use `renderToString`, so we can handle the dependency ourselves, without throwing a warning for people who won't use the method

**Comment by dotnetCarpenter** [comment]
@eddyerburgh That was not what I intended. If you have an API that has `renderToString` it should work out of the box. Having *vue-server-renderer* as a peer dependency is good as it gives me the choice to install it either as a `devDependecy` or a `dependency`.

I have a *laravel-mix* project that does not use SSR, hence do not need *vue-server-renderer* and I got a nuxt project that already has *vue-server-renderer*. I prefer nuxt to handle versions of *vue-server-renderer* but in my *laravel-mix* project, I have installed it as a `devDependecy` because it is only relevant to my tests.

But if you want to put in the work to handle it in *vue-test-utils*, that is of course fine. But I was not aiming at that. 😄 I was just expecting a better explanation to why I have to install a package, I am not using. I think a small update to the release text will suffice.

**Pull request merged**